test/policies/agentPolicyTests.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import "chai/register-should";
+import { AgentSettings } from "../../lib/serviceClient";
+import { RequestPolicyOptions } from "../../lib/policies/requestPolicy";
+import { WebResource, WebResourceLike } from "../../lib/webResource";
+import { HttpHeaders } from "../../lib/httpHeaders";
+import { agentPolicy, AgentPolicy } from "../../lib/policies/agentPolicy";
+import { nodeDescribe, browserDescribe } from "../msAssert";
+
+describe("AgentPolicy", function () {
+  const emptyRequestPolicy = {
+    sendRequest: (_: WebResourceLike) =>
+      Promise.resolve({
+        request: new WebResource(),
+        status: 404,
+        headers: new HttpHeaders(undefined),
+      }),
+  };
+
+  const emptyPolicyOptions = new RequestPolicyOptions();
+
+  nodeDescribe("for Node.js", function () {
+    const http = require("http");
```

Comment:
The intent here is to ensure that there libraries only get loaded running in Node environments, hence the inline `require` use.